### ISSUE TITLE
Customizer Textbox support

### ISSUE THREAD

**Issue by ThePilcrow** [openscad/openscad#5067]
Stated is in the Customizer manual section "The text box example only works in release version 2021.01, it may not work in future versions!"

Indeed in the OpenSCAD-2024.03.23 Development Snapshots version this option is no longer working.
The result is that certain options, that is a vector with more than 4 elements, can no longer be configured.

The problem, or rather the bug, is that there is no alternative way to configure a vector with more than 4 elements.
Hence, at least my design( https://www.printables.com/model/294305-storage-box-generator-v100 ) can no longer be used!

Thus, provide a way to configure/use vectors with more than 4 elements in the Customizer.

**Comment by jordanbrown0** [comment]
This is a subset of #3817, but since it might be simpler than the full nested-vector support requested there I'll leave it open.

**Comment by gchwelos** [comment]
+1 for similar purpose.

**Comment by vingenuity** [comment]
I would also like to put in a vote for this feature.

I just tried out the latest version of OpenSCAD and would like to continue using it, but this is a pretty substantial regression. I am not going to keep multiple versions of the software to export certain models.

**Comment by gsohler** [comment]
I am interested in:

Whats the use case to have vectors with 4+ elements ?
Are you designing Tesseracts  in Minkowski Space ?

**Comment by ThePilcrow** [comment]
@gsohler: It has nothing to do with vectors but there is no alternative for passing an array of values using the Customizer.

**Comment by gsohler** [comment]
I tried this code:

function get_n(x,n)=x[n];

a=["a","b","c","d","e"];

echo(get_n(a,3));


And it worked as expected. Where does passing a vector values fail ?

**Comment by ThePilcrow** [comment]
Please read my original message. I did not state that passing a vector fails, in version 2021.01
In the manual of version 2012.01 it is stated that the Text Box in future versions will not work anymore. And indeed in version OpenSCAD-2024.03.23 the TextBox option is not available anymore. The TextBox is what I am using in the Customizer for people to enter certain values.
Check out my design( https://www.printables.com/model/294305-storage-box-generator-v100 ). It works under version 2021.01 but can no longer be used! In the Customizer look at the section Drawer Configuration, last option _compartmentArrangement_. As you can see in the source code there should be a TextBox called _compartments_ which is no longer shown. Same for Drawer Configuration Labels, under _labelArrangement_ a field called called _labels_ should be present.
Thus, what is or will be the alternative for a TextBox.

**Comment by vingenuity** [comment]
> I am interested in:
> 
> Whats the use case to have vectors with 4+ elements ?
> Are you designing Tesseracts  in Minkowski Space ?
> […](#)

No. In this case, "vector" is being used as the C++ data structure term (i.e. array), not the mathematical term. Being able to customize a set of more than four values seems pretty useful in my opinion.

For a more concrete example, I have a SCAD file that allows customization of cups by accumulating a set of [diameter, height] values to create a cylinder:
```
/* [Cup] */
// Diameters of the cup at each height, starting at zero.
Cup_Diameters = [66.0, 71.0, 86.0, 89.0, 91.0, 99.0];

// Heights between each of the provided diameters. Should have one less than the diameter count.
Cup_Heights = [30.0, 79.0, 9.0, 8.0, 12.0];
```
And yes, this specific example would be better served by the 2D vectors of #3817; however, until that solution is in place, removing the original (if flawed) version of vector customization leaves me with no way to work on this SCAD file.

**Comment by illigtr** [comment]
I am interested in this as well. I have vectors with 6 or 8 elements that do not work with any nightly build. It appears a hard limit of 4 has been introduced.
